src/main/java/org/apache/mesos/offer/ResourceCleanerScheduler.java

```diff
@@ -17,13 +18,39 @@
   public ResourceCleanerScheduler(
       ResourceCleaner resourceCleaner,
       OfferAccepter offerAccepter) {
-
     this.resourceCleaner = resourceCleaner;
     this.offerAccepter = offerAccepter;
   }
 
   public List<OfferID> resourceOffers(SchedulerDriver driver, List<Offer> offers) {
-    List<OfferRecommendation> recommendations = resourceCleaner.evaluate(offers);
-    return offerAccepter.accept(driver, recommendations);
+    final List<OfferRecommendation> recommendations = resourceCleaner.evaluate(offers);
+    final Map<Protos.SlaveID, List<OfferRecommendation>> recommendationsGroupedByAgents =
```

Comment:
Add an explanation for why these need to be grouped before passing to the OfferAccepter? does this affect the accept logic?